src/modules/navigator/mission.cpp

```diff
@@ -519,13 +519,17 @@ Mission::set_mission_items()
 			/* use current mission item as next position item */
 			memcpy(&mission_item_next_position, &_mission_item, sizeof(struct mission_item_s));
 			has_next_position_item = true;
-			float altitude = _navigator->get_global_position()->alt;
-			if (pos_sp_triplet->current.valid) {
-				altitude = pos_sp_triplet->current.alt;
+
+			/* hold current altitude when no altitude is given */
+			if (_mission_item.altitude < 0.5f) {
```

Comment:
So if I fly at sea level an altitude request of 0 meters will be rejected? This doesn't look completely thought out.